jgrapht-core/src/main/java/org/jgrapht/graph/AbstractBaseGraph.java

```diff
@@ -39,32 +50,55 @@
  * @param <E> the graph edge type
  *
  * @author Barak Naveh
+ * @author Dimitrios Michail
  * @since Jul 24, 2003
  */
 public abstract class AbstractBaseGraph<V, E>
-    extends
-    AbstractGraph<V, E>
-    implements
-    Graph<V, E>,
-    Cloneable,
-    Serializable
+    extends AbstractGraph<V, E>
+    implements Graph<V, E>, Cloneable, Serializable
 {
     private static final long serialVersionUID = 4811000483921413364L;
 
     private static final String LOOPS_NOT_ALLOWED = "loops not allowed";
     private static final String GRAPH_SPECIFICS_MUST_NOT_BE_NULL =
         "Graph specifics must not be null";
 
-    private EdgeFactory<V, E> edgeFactory;
     private transient Set<V> unmodifiableVertexSet = null;
 
+    private Supplier<V> vertexSupplier;
+    private Supplier<E> edgeSupplier;
+    
+    @Deprecated
+    private EdgeFactory<V, E> edgeFactory;
+
+    private GraphType type;
```

Comment:
I think you either have to write a custom deserializer for type in terms of the old deleted fields, or else modify serialVersionUID to indicate that graphs serialized by previous versions cannot be deserialized by this version.